lib/connections/service.go

```diff
@@ -370,23 +378,45 @@ func (s *Service) connect() {
 	}
 }
 
-func (s *Service) shouldLimit(addr net.Addr) bool {
-	if s.cfg.Options().LimitBandwidthInLan {
+func (s *Service) shouldLimit(addr net.Addr, options config.OptionsConfiguration) bool {
+	if options.LimitBandwidthInLan {
 		return true
 	}
 
 	tcpaddr, ok := addr.(*net.TCPAddr)
 	if !ok {
 		return true
 	}
-	for _, lan := range s.lans {
+
+	lans := s.determineLocalNetworks(options.AlwaysLocalNets)
+	for _, lan := range lans {
 		if lan.Contains(tcpaddr.IP) {
 			return false
 		}
 	}
 	return !tcpaddr.IP.IsLoopback()
 }
 
+func (s *Service) determineLocalNetworks(alwaysLocalNets []string) []*net.IPNet {
+	// This will be used on connections created in the connect and listen routines.
+	lans, _ := osutil.GetLans()
```

Comment:
These were cached before, not we're hitting them on every incoming outgoing connection which is expensive.